### DISCUSSION TITLE
"Current Offset lag interval" metric

### DISCUSSION THREAD

**Discussion by saloni-2404** [strimzi/strimzi-kafka-operator#4578] [Q&A]
Hi Jakub,
Would we be able to get this metric "Current Offset lag interval" by Strimzi kafka operator?
I tried curling kafka-exporter and could not find in the output. 
Should we be enabling any parameter in the kafka.yaml for this?
Thanks

**Comment by scholzj** [comment]
I do not think Kafka provides this metric. This is normally provided only the the Kafka Exporter.

**Reply by saloni-2404** [reply]
ok, somehow, i am not able to curl this metric from kafka-exporter endpoint as well

**Reply by scholzj** [reply]
So how are you looking for it? What are you searching for and where?

**Reply by saloni-2404** [reply]
So we have a monitoring pod setup, from which we are using the 9404 port to extract the metrics to prometheus, so go inside this monitoring pod, and i do a 
curl http://my-cluster-kafka-exporter:9404/metrics
I get a bunch of metrics, but not this one in particular, "Current Offset lag interval"

**Reply by scholzj** [reply]
Well, maybe your monitoring pod does nto collect it? Have you tried to do it directly in the Kafka Exporter pod? Also, what metric are you looking for?

**Reply by saloni-2404** [reply]
Looking for current_offset_lag_interval,
Could you please help point to where to look inside kafka-exporter pod

**Reply by scholzj** [reply]
Well, you just exec there and do `curl localhost:9404/metrics` and grep through it.

**Reply by saloni-2404** [reply]
ok did that, i do not see the metric i was mentioning above..

**Reply by scholzj** [reply]
And do you actually have some consumers / consumer groups consuming things?

**Reply by scholzj** [reply]
BTW: Why do you think it should be `current_offset_lag_interval`? I seem to have the consumer lag as `kafka_consumergroup_lag`.

```
$ curl -s localhost:9404/metrics | grep kafka_consumergroup_lag
# HELP kafka_consumergroup_lag Current Approximate Lag of a ConsumerGroup at Topic/Partition
# TYPE kafka_consumergroup_lag gauge
kafka_consumergroup_lag{consumergroup="__strimzi-topic-operator-kstreams",partition="0",topic="__strimzi_store_topic"} 0
kafka_consumergroup_lag{consumergroup="my-group",partition="0",topic="kafka-test-apps"} 0
kafka_consumergroup_lag{consumergroup="my-group",partition="1",topic="kafka-test-apps"} 0
kafka_consumergroup_lag{consumergroup="my-group",partition="2",topic="kafka-test-apps"} 0
kafka_consumergroup_lag{consumergroup="my-kafka-consumer",partition="0",topic="kafka-test-apps"} 0
kafka_consumergroup_lag{consumergroup="my-kafka-consumer",partition="1",topic="kafka-test-apps"} -1
kafka_consumergroup_lag{consumergroup="my-kafka-consumer",partition="2",topic="kafka-test-apps"} 0
```

**Reply by saloni-2404** [reply]
Yes, this metric, im able to see too, kafka_consumergroup_lag, but the requirement is this metric: "Current Offset lag interval"

**Reply by scholzj** [reply]
Well, who gave you such requirement? Maybe such metric does not exist?

**Reply by scholzj** [reply]
Or maybe more general ... what do you expect this metric to be about? Why do you think the Kafka Exporter supports it?

**Reply by saloni-2404** [reply]
ok, when i try to check the definition for this requirement it says:

Provides a delta from the last reported metric value. When this metric has a zero value for a few intervals, the consumer group has stopped reading messages or has become inactive.

**Reply by saloni-2404** [reply]
would that be same as our consumergroup_lag?

**Reply by scholzj** [reply]
I don't know ... probably not exactly the same. Honestly not sure I understand what I would expect under that description.

**Reply by scholzj** [reply]
So back to the other question ... why do you think this metric should be in the Kafka Exporter?
